utils/src/main/scala/it/pagopa/interop/commons/utils/errors/AkkaResponses.scala

```diff
@@ -11,21 +11,24 @@ import it.pagopa.interop.commons.utils.errors.GenericComponentErrors.GenericErro
 
 trait AkkaResponses {
 
-  private def completeWithError(statusCode: StatusCode, error: ComponentError)(implicit
-    contexts: Seq[(String, String)],
-    serviceCode: ServiceCode
-  ): StandardRoute = complete(statusCode.intValue, Problem(statusCode, error, serviceCode, getCorrelationId(contexts)))
-
-  private def completeWithErrors(statusCode: StatusCode, errors: List[ComponentError])(implicit
-    contexts: Seq[(String, String)],
-    serviceCode: ServiceCode
-  ): StandardRoute = complete(statusCode.intValue, Problem(statusCode, errors, serviceCode, getCorrelationId(contexts)))
-
   private def completeWithError(statusCode: StatusCode, headers: List[HttpHeader], error: ComponentError)(implicit
     contexts: Seq[(String, String)],
     serviceCode: ServiceCode
-  ): StandardRoute =
-    complete(statusCode.intValue, headers, Problem(statusCode, error, serviceCode, getCorrelationId(contexts)))
+  ): StandardRoute = headers match {
+    case Nil => complete(statusCode.intValue, Problem(statusCode, error, serviceCode, getCorrelationId(contexts)))
+    case _   =>
+      complete(statusCode.intValue, headers, Problem(statusCode, error, serviceCode, getCorrelationId(contexts)))
+  }
```

Comment:
Any reason why we do not pass the empty header?
```suggestion
  ): StandardRoute = complete(statusCode.intValue, headers, Problem(statusCode, error, serviceCode, getCorrelationId(contexts)))
```